<?xml version="1.0" encoding="UTF-8"?>
<record_update table="sysauto_script">
    <sysauto_script action="INSERT_OR_UPDATE">
        <active>true</active>
        <condition/>
        <conditional>false</conditional>
        <name>Self-Service Update Executions</name>
        <run_as display_value="System Administrator">6816f79cc0a8016401c5a33be04be441</run_as>
        <run_as_tz/>
        <run_dayofmonth>1</run_dayofmonth>
        <run_dayofweek>1</run_dayofweek>
        <run_period>1970-01-01 00:01:00</run_period>
        <run_start>2015-01-23 14:02:25</run_start>
        <run_time>1970-01-01 08:00:00</run_time>
        <run_type>periodically</run_type>
        <script><![CDATA[var authenticator = new RightScaleAuthenticator();
var helper = new RightScaleHelper();
var ready = helper.readyForUpdate(current.run_start, 'execution');
var forceUpdateMessage = "";
var account = new GlideRecord('x_ris_rightscale_c_accounts');
account.query();
while (account.next()) {
	if (account.executions_force_update) {
		forceUpdateMessage = "Force ";
		account.executions_force_update = false;
		account.update();
	} else if (!ready) {
		continue;
	}
	gs.info("Executing Scheduled Job: Self-Service " + forceUpdateMessage +  "Update Executions for account " + account.u_id);
	if (!authenticator.validateAccessToken(account)){
		if (!authenticator.updateAccountAccessToken(account)) continue;
	}
	
	try {
		// login to SS every time to make sure a session is not expired
		var responseState = authenticator.loginToSelfService(account);
		if (responseState != 200) {
			gs.error("Updating Executions failed for account " + account.u_id + " due to unsuccessful login to Self-Service.");
			continue;
		}
		var executions = new RightScaleResource('executions', 'manager');
		var ssExecutions = executions.getSSResource(account);
		if (!ssExecutions) {
			gs.error("Updating Executions failed for account " + account.u_id + " due to unsuccessful response to RS API.");
			continue;
		}
		executions.updateAll(ssExecutions, account);
	} catch(ex) {
		gs.error("Updating Executions failed for account " + account.u_id + " with error: " + ex.toString());
	}
}]]></script>
        <sys_class_name>sysauto_script</sys_class_name>
        <sys_created_by>admin</sys_created_by>
        <sys_created_on>2015-02-20 14:55:52</sys_created_on>
        <sys_id>bfc6ac660f96b500bad2fc5be1050eb4</sys_id>
        <sys_mod_count>65</sys_mod_count>
        <sys_name>Self-Service Update Executions</sys_name>
        <sys_package display_value="RightScaleCloudManagement" source="x_ris_rightscale_c">adee18db0f61f100bad2fc5be1050e29</sys_package>
        <sys_policy/>
        <sys_scope display_value="RightScaleCloudManagement">adee18db0f61f100bad2fc5be1050e29</sys_scope>
        <sys_update_name>sysauto_script_bfc6ac660f96b500bad2fc5be1050eb4</sys_update_name>
        <sys_updated_by>admin</sys_updated_by>
        <sys_updated_on>2015-12-03 12:10:55</sys_updated_on>
        <upgrade_safe>false</upgrade_safe>
    </sysauto_script>
</record_update>
